pylint/pyreverse/utils.py

```diff
@@ -60,37 +67,18 @@ def get_visibility(name):
     return visibility
 
 
-ABSTRACT = re.compile(r"^.*Abstract.*")
-FINAL = re.compile(r"^[^\W\da-z]*$")
-
-
-def is_abstract(node):
-    """Return true if the given class node correspond to an abstract class
-    definition.
-    """
-    return ABSTRACT.match(node.name)
-
-
-def is_final(node):
-    """Return true if the given class/function node correspond to final
-    definition.
-    """
-    return FINAL.match(node.name)
-
-
-def is_interface(node):
+def is_interface(node: nodes.ClassDef) -> bool:
     # bw compatibility
     return node.type == "interface"
```

Comment:
So, here, mypy thinks we return "Any" ?!

Reply:
Yes, no clue why. Maybe because the `type` attribute is monkey-patched on the `node` class.
But even VS Code recognises `node.type` correctly as `str`.

Reply:
https://github.com/python/mypy/issues/5697#issuecomment-425738430

Reply:
Thank you, that explains why this happens here.
On lines 213 and 215 I still don't know why `mypy` complains here:
```python
def get_annotation_label(ann: nodes.Name | nodes.NodeNG) -> str:
    if isinstance(ann, nodes.Name) and ann.name is not None:
        return ann.name. # <-- l.213
    if isinstance(ann, nodes.NodeNG):
        return ann.as_string()  # <-- l.215
    return ""
```
Using `typing.reveal_type()`, both `ann.name` and `ann.as_string()` are recognised as `str`, not `Any`.

Reply:
Although we have already added some annotations to astroid, mypy will not use them until we add a `py.typed` file (once they are done). Until then all astroid types are basically inferred as `Any`.

Reply:
https://github.com/PyCQA/astroid/issues/1287